Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -3351,40 +3351,81 @@ - (CABasicAnimation *)_animationForFindOverlay:(BOOL)animateIn
     return animation;
 }
 
-- (void)_updateFindOverlayPosition
+- (void)_updateFindOverlayForOverflowScrollPositions
 {
+    if (_findOverlaysForOverflowScroll.isEmpty())
+        return;
+
     UIScrollView *scrollView = _scrollView.get();
-    [_findOverlay setBounds:CGRectMake(0, 0, scrollView.bounds.size.width, scrollView.bounds.size.height)];
-    [_findOverlay setCenter:CGPointMake(scrollView.center.x + scrollView.contentOffset.x, scrollView.center.y + scrollView.contentOffset.y)];
+    auto topView = _findOverlaysForOverflowScroll[FindOverlayView::Top].get();
+    auto rightView = _findOverlaysForOverflowScroll[FindOverlayView::Right].get();
+    auto bottomView = _findOverlaysForOverflowScroll[FindOverlayView::Bottom].get();
+    auto leftView = _findOverlaysForOverflowScroll[FindOverlayView::Left].get();
```

Comment:
Our smart pointer guidelines suggest these should remain as smart pointers on the stack, so I think we prefer:

```suggestion
    RetainPtr topView = _findOverlaysForOverflowScroll[FindOverlayView::Top];
    RetainPtr rightView = _findOverlaysForOverflowScroll[FindOverlayView::Right];
    RetainPtr bottomView = _findOverlaysForOverflowScroll[FindOverlayView::Bottom];
    RetainPtr leftView = _findOverlaysForOverflowScroll[FindOverlayView::Left];
```